src/cpu/reg_class.hh

```diff
@@ -354,15 +360,30 @@ RegClass::operator[](RegIndex idx) const
     return RegId(*this, idx);
 }
 
+// Type matching for gem5::VecRegContainer class
+// This is used in TypedRegClassOps.
+template<typename>
+struct is_vec_reg_container : std::false_type {};
+template<std::size_t SIZE>
+struct is_vec_reg_container<gem5::VecRegContainer<SIZE>> : std::true_type {};
+
 template <typename ValueType>
 class TypedRegClassOps : public RegClassOps
 {
   public:
     std::string
     valString(const void *val, size_t size) const override
     {
-        assert(size == sizeof(ValueType));
-        return csprintf("%s", *(const ValueType *)val);
+        if constexpr (is_vec_reg_container<ValueType>::value) {
+            if (size == sizeof(ValueType)) {
+                return csprintf("%s", *(const ValueType *)val);
+            } else {
+                return ((const ValueType *)val)->getString(size);
+            }
+        } else {
+            assert(size == sizeof(ValueType));
```

Comment:
Should this be a static assert?

Reply:
Yeah, agreed. The value passed to size is regBytes(), which is a constexpr.

Reply:
I just figure out that making this assert being a static_assert does not work because of this change.

Previously, there is only one place calling this function, and the value passed to `size` is always a constexpr, so the static_assert would make sense.

Now that we call this function is called at two different places, thus `size` is no longer guaranteed to be determined at compilation time.